client/my-sites/importer/section-import.jsx

```diff
@@ -78,23 +78,27 @@ class SectionImport extends Component {
 	state = getImporterState();
 
 	onceAutoStartImport = once( () => {
-		const { engine, site } = this.props;
-		const { importers: imports } = this.state;
+		const { engine, site, afterStartImport } = this.props;
+		const { importers } = this.state;
 
 		if ( ! engine ) {
 			return;
 		}
 
-		if ( ! isEmpty( imports ) ) {
-			// Never clobber an existing import
-			return;
-		}
-
-		if ( ! importerComponents[ engine ] ) {
-			return;
+		// If there is no existing import and the `engine` is valid, start a new import.
+		if ( isEmpty( importers ) && importerComponents[ engine ] ) {
+			defer( () => {
+				startImport( site.ID, getImporterTypeForEngine( engine ) );
+				// After import was started, redirect back to the route without `engine` query arg.
+				// That removes the `engine` prop from this component and doesn't spoil future
+				// rendering when the import is, e.g., cancelled.
+				afterStartImport?.( true );
```

Comment:
Looking at the definition of `afterStartImport` above, it doesn't seem to have an argument. What did we aim to achieve with the `true` here and `false` below?

Reply:
It says whether the import was really started (`true`) or not. We can choose to ignore the request when the `engine` argument has an unsupported value or when an existing import is already running. The next step is the same in both cases: consider the request done and redirect back to URL without the query args.

I don't really use the argument, but it looked like a good thing to have 🙂

Reply:
Makes sense to me 👍  I generally remove any unused pieces (YAGNI FTW), but I guess you're envisioning a scenario where this argument would actually be used?

Reply:
OK, I pushed a commit that removes the `afterStartImport` arg because I don't really have any use for it neither now nor in the future. It was just something of a thought experiment 🙂